src/__tests__/spectral.test.ts

```diff
@@ -177,6 +300,73 @@ Array [
     expect(results.length).toEqual(1);
   });
 
+  // Assures: https://stoplightio.atlassian.net/browse/SL-788
+  test('run with rulesets overrides ruleset on run, not permenantly', () => {
+    const spec = {
+      hello: 'world',
+    };
+
+    const rulesets: IRuleset[] = [
+      {
+        rules: {
+          format: {
+            test: {
+              type: RuleType.STYLE,
+              function: RuleFunction.TRUTHY,
+              path: '$',
+              enabled: false,
+              severity: RuleSeverity.ERROR,
+              summary: '',
+              input: {
+                properties: 'nonexistant-property',
+              },
+            },
+          },
+        },
+      },
+    ];
+
+    const overrideRulesets: IRuleset[] = [
+      {
+        rules: {
+          format: {
+            test: true,
+          },
+        },
+      },
+    ];
+
+    const s = new Spectral({ rulesets });
+
+    // run again with an override config
+    const run1 = s.run({ target: spec, spec: 'format', rulesets: overrideRulesets });
+
+    expect(s.getRules('format')).toMatchInlineSnapshot(`
```

Comment:
You may want to consider checking this:

```suggestion
    const s = new Spectral({ rulesets });
    const originalRules = merge({}, s.getRules('format'));
    
    const run1 = s.run({ target: spec, spec: 'format', rulesets: overrideRulesets });
    
    expect(s.getRules('format')).toEqual(originalRules);
```

Would you agree that such check would express the intention (of not overriding the original ruleset) more immediately?

Reply:
Agreed